File: P2-Integration/rs-json-openapi/src/main/python/consumer/consumer.py
from swagger_client import CalculatorApi

calculator = CalculatorApi()
x, y = 7.0, 5.0

result = calculator.add(x, y)
print(str(x) + ' + ' + str(y) + ' = ' + str(result))

result = calculator.subtract(x, y)
print(str(x) + ' - ' + str(y) + ' = ' + str(result))

result = calculator.multiply(x, y)
print(str(x) + ' x ' + str(y) + ' = ' + str(result))

result = calculator.divide(x, y)
print(str(x) + ' : ' + str(y) + ' = ' + str(result))

result = calculator.compute_all(x, y)
print(result)

File: P2-Integration/case-studies/xs-calculator/src/main/java/ma/aui/sse/paradigms/integration/xs/consumer/Consumer.java
package ma.aui.sse.paradigms.integration.xs.consumer;

import ma.aui.sse.paradigms.integration.xs.consumer.stub.*;

public class Consumer {
    
    public static void main(String[] args) {

        // calculator is just a proxy, a stub object -- No business implementation
        // It's a fake object giving the impression and luxury to the the consumer
        // that it's using the remote business (real) implementation as if it were local
        Calculator calculator = new CalculatorService().getCalculatorPort();

        double x = 7.0, y = 5.0, result;

        // Remote calls (as if they were local!)
        // No sockets, no protocol, no hassle!
        // All is hidden behind the scenes
        result = calculator.add(x, y);
        System.out.println(x + " + " + y + " = " + result);

        result = calculator.subtract(x, y);
        System.out.println(x + " - " + y + " = " + result);

        result = calculator.multiply(x, y);
        System.out.println(x + " x " + y + " = " + result);

        result = calculator.divide(x, y);
        System.out.println(x + " : " + y + " = " + result);

        Result res = calculator.computeAll(x, y);
        System.out.println("Sum: " + res.getSum());
        System.out.println("Difference: " + res.getDifference());
        System.out.println("Product: " + res.getProduct());
        System.out.println("Ratio: " + res.getRatio());
    }
}